### PR TITLE
Front end for donations

### PR DESCRIPTION
**Pull request by erib3**

### CODE REVIEW

**Comment by makoto** [comment]
This is for https://github.com/wearekickback/app/issues/206 I presume

**Comment by jefflau** [comment]
Thanks for this @erib3! We're currently working to switch all components to functional components. For old components we can accept the use of classes, so you can leave that. But can you please refactor your new components to functional components with hooks.

Thanks

**Comment by erib3** [comment]
Ok no problem, I made the change. I look forward to getting your feedback.

File: src/components/SingleEvent/TipOrganiser.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}) => [
+  <p>How much would you like to tip the organiser?</p>,
+  <div onChange={changeTipAmount.bind(this)}>
+    <input type="radio" value={all} name="tip" /> All ({all}) ETH
+    <input type="radio" value={extra} name="tip" /> Extra ({extra} ETH )
+    <input type="radio" value={coffee} name="tip" /> Coffee ({coffee}) ETH
+  </div>,
+  <br></br>,
+  <p>
+    Total tip: {tip} ETH (and withdraw {withdraw} ETH)
```
Comment (makoto) [inline src/components/SingleEvent/TipOrganiser.js]: `works for both eth and dai` does not look like it works on DAI

File: src/components/SingleEvent/EventCTA.js [hunk 2]
```diff
@@ -97,11 +118,43 @@ class EventCTA extends Component {
...
+      return <WithdrawPayout address={address} amount={myShare} />
+    }
+    if (this.state.mode === donate) {
+      var all = myShare
+      var extra =
+        myShare -
+        toEthVal(deposit)
+          .toEth()
+          .toFixed(3)
+      var coffee = 0.01
```
Comment (makoto) [inline src/components/SingleEvent/EventCTA.js]: You can't just hard code 0.01 eth . You need to hit external price api to get the price .

**Comment by makoto** [comment]
@erib3 the requirement is to hook up with mutation which you haven't done it.

**Comment by erib3** [comment]
@makoto just updated, does the coffee price calculation and new resolver look ok?

**Comment by erib3** [comment]
Ok @jefflau I added a component that's modeled after WithdrawPayout. Let me know how it looks, Im still new to react

File: src/components/SingleEvent/EventCTA.js [hunk 3]
```diff
@@ -97,11 +131,53 @@ class EventCTA extends Component {
...
+      return <WithdrawPayout address={address} amount={myShare} />
+    }
+    if (this.state.mode === Donate) {
+      const Extra =
+        myShare -
+        toEthVal(deposit)
+          .toEth()
+          .toFixed(3)
+      var coffee = currencySymbol === 'DAI' ? 3 : this.getCoffeePrice()
+      //TODO: add in address
```
Comment (makoto) [inline src/components/SingleEvent/EventCTA.js]: What is this? What do you have to do to finish it off?

**Comment by makoto** [comment]
@erib3 as we discussed, we will close this PR as the requirement wasn't fully met.
I instructed Gitcoin to make the agreed partial payment.

Here is the feedback of this PR of why it is not complete, so that anyone who will take this over can take into consideration.

## Feedback on this current submission

- Add `console.log('sendAndWithdrawPayout resolver function is called with', {addresses, values})` at https://github.com/wearekickback/app/pull/211/files#diff-0d76ee09d2dc74150400cda12b47dddbR263 so that we know for sure that the correct resolver function is called.
- When showing "Withdraw payout" button, please also add "Or tip organiser" link so that user can go back and fourth between two actions.
- Do not show "Coffee" option if total amount to withdraw is less than Coffee amount ($3)
- Do not show "Extra" option if there is no extra (= 100% attended).
- Please follow [our contribution style guide](https://github.com/wearekickback/app#dev-guide)
- When raising PR, please fill in our [pull request template](https://github.com/wearekickback/app/blob/dev/pull_request_template.md)

## Outstanding requirement

- Correct amount of "Coffee" for ETH is not calculated.
- Default is not selected (please select "Coffee" by default)
- Links ("Or Withdraw all", "Go back") pointer do not change when hovered.
- When "Tip to Withdraw" is clicked, it does withdraw.
- Mutation definition for sendAndWithdrawPayout is missing

**Pull request closed without merging**